sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -144,13 +145,21 @@ class QueryExecution(
       )
       append("\n")
     } catch {
-      case _: AnalysisException =>
+      case e: AnalysisException => analysisException = e.toString
+    }
+    if (analysisException.size > 0) {
+      QueryPlan.append(analyzed, append, verbose, addSuffix, maxFields)
```

Comment:
Statements for if and else seems to be swapped.

Reply:
If we move this to try-catch (as @cloud-fan suggested), we don't need if-else.

Reply:
Sure.